index.js

```diff
@@ -1,32 +1,75 @@
 const debug = require('debug')('hub-sync');
 const HubSync = require('./sync');
+const readline = require('readline');
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stderr
+});
 
-const readToken = ({ token, tokenFile }) => {
+const readToken = async ({ token, tokenFile }) => {
   if (token) {
     return token;
   }
   try {
-    return fs.readFileSync(tokenFile, 'utf-8').trim();
+    try {
+      await fs.promises.access(tokenFile, fs.constants.R_OK);
+    } catch {
+      console.error(chalk`{magenta hub-sync needs a {bold GitHub Personal Access Token} to update your GitHub repo fork.}`);
+      console.error(chalk`Please visit {underline https://github.com/settings/tokens} and follow the instructions:`);
+      console.error(chalk`1. {italic If asked}, type your GitHub account password`);
+      console.error(chalk`2. Click the {green {italic Generate new token}} button.`);
+      console.error(chalk`3. Type "{cyan hub-sync}" in the {yellow {italic Note}} input box.`);
+      console.error(chalk`4. In the {yellow {italic Select scopes}} section, locate {yellow {italic repo}} subsection and select "{green public_repo}".`);
+      console.error(chalk`5. Click the {green {italic Generate token}} button at the very bottom of the page.`);
+      console.error(chalk`{red {bold Notice: Keep you Personal Access Token ABSOLUTELY CONFIDENTIAL}}.`);
+      console.error(chalk`6. Copy-paste the token here:`);
+      console.error(chalk`(your token will be saved to {yellow ${tokenFile}}, so you don't need to generate token every time you update your fork.)`);
+      const res = await new Promise((resolve) => {
+        rl.question('Your token: ', (input) => { resolve(input.trim()); } );
+      });
+      if (!/^[0-9a-f]{40}$/.test(res)) {
+        console.error(chalk`{red It doesn't look like a valid token. Give up.}`);
+        return undefined;
+      }
+      console.error(chalk`{magenta Token recieved, trying to save to} {yellow ${tokenFile}}`);
+      await fs.promises.writeFile(tokenFile, res, 'utf-8');
+      console.error(chalk`{green Token saved to} {yellow ${tokenFile}} successfully`);
+      try {
+        await fs.promises.chmod(tokenFile, 0o600);
+      } catch (e) {
+        console.error(chalk`{red Cannot change the mode of} {yello ${tokenFile}} {red to 0600. You need to sure the security of the file by yourself.}`);
+      }
+    }
+    return (await fs.promises.readFile(tokenFile, 'utf-8')).trim();
   } catch (e) {
-    console.error(`Error occured during reading token file ${tokenFile}.`);
+    console.error(chalk`{red Error occured when accessing token file} {yellow ${tokenFile}}.`);
     console.error('Please check its existance and permission.');
+    console.error('Also make sure if the folder it belongs to actually exist.');
+    console.error(chalk.dim(e.stack));
     return undefined;
   }
 }
 
-const parseSpec = (spec = '', omitUser = false) => {
+const parseSpec = (spec, dest) => {
```

Comment:
So... what does the `spec` in the `parseSpec()` do?

Reply:
`spec` is a string representation of a user-repo-branch tuple. A valid `hub-sync` invocation contains two `spec`s, denoting *which user/repo/branch tuple is to be update* and *from which user/repo/branch is to be used as reference*.

Reply:
Understood. I guessed as what you said. I just got little confused in the naming QuQ